src/Raven.Server/Documents/Handlers/Processors/AbstractStatsHandlerProcessor.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Linq;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations;
+using Raven.Server.Documents.Indexes;
+using Raven.Server.Documents.Sharding;
+using Raven.Server.Web;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors
+{
+    internal abstract class AbstractStatsHandlerProcessor<TRequestHandler, TOperationContext> : AbstractHandlerProcessor<TRequestHandler, TOperationContext>
+        where TRequestHandler : RequestHandler
+        where TOperationContext : JsonOperationContext
+    {
+        protected AbstractStatsHandlerProcessor([NotNull] TRequestHandler requestHandler, [NotNull] JsonContextPoolBase<TOperationContext> contextPool) : base(requestHandler, contextPool)
+        {
+        }
+
+        protected IndexInformation[] GetDatabaseIndexesFromRecord()
+        {
+            if (RequestHandler is ShardedRequestHandler shardedRequestHandler)
+            {
```

Comment:
the sharded probably should simply override this?

Reply:
actually `GetDatabaseIndexesFromRecord` should be only at the sharded, no?

Reply:
okay, I realize there is a problem with the inheritance and we can't have a common abstract class..
wouldn't it be better to hook such things on the `ShardedContext` ? an extension for the  `record.Indexes` ?